src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.client.core;
+
+import java.io.Serializable;
+
+/**
+ * This class defines all non-static parameters needed to create an HttpClient object. It is used as
+ * the key for the static hashmap of reusable http clients.
+ */
+public class HttpClientSettingsKey implements Serializable {
+
+  private OCSPMode ocspMode;
+  private boolean useProxy;
+  private String proxyHost = "";
+  private int proxyPort = 0;
+  private String nonProxyHosts = "";
+  private String proxyUser = "";
+  private String proxyPassword = "";
+
+  public HttpClientSettingsKey(
+      OCSPMode mode, String host, int port, String nonProxyHosts, String user, String password) {
+    this.useProxy = true;
+    this.ocspMode = mode != null ? mode : OCSPMode.FAIL_OPEN;
+    this.proxyHost = host.trim();
+    this.proxyPort = port;
+    this.nonProxyHosts = nonProxyHosts.trim();
+    this.proxyUser = user.trim();
+    this.proxyPassword = password.trim();
```

Comment:
Do we want to check whether the input String is null?

Reply:
Yes! Changed! I was checking for null before instantiating the HttpClientSettingsKey object, but it makes more sense to check for null here instead for future use cases.